content/blog/2022-07-05-Practical-data-drift-monitoring-with-DVC.md

```diff
@@ -0,0 +1,337 @@
+---
+title: 'Practical data drift monitoring with DVC'
+date: 2022-07-05
+description: >
+  Showing how to implement data drift detection with DVC.
+
+descriptionLong: >
+  Detection of data drift has recently been put in the spotlight as a very
+  important step in the ML system, as we already discussed in [Preventing Stale
+  Models in Production](https://dvc.org/blog/stale-models). We will go deeper in
+  this blog post and set up an automatic drift detector as part of the ML
+  pipeline. Quite a few libraries, even open-source, provide  users with drift
+  detection capabilities. We will show how we may integrate  them into the ML
+  system with DVC and how, in general, DVC features are  beneficial for data
+  drift detection.
+picture: TODO
+author: dan_martinec
+commentsUrl: TODO
+
+tags:
+  - Data Validation
+  - Machine Learning
+  - MLOps
+  - Productionization
+---
+
+# Introduction
+
+## Data distribution shift
+
+There is a large boom of ML systems deployed to production these days. This
+brings many demands to ML engineers. ML pipeline automation is possibly the most
+important one. However, there is also one less known but very important aspect.
+That is the validation of inputs and outputs of the ML system. In fact, data
+validation is listed as one of the
+[hidden technical debts in machine learning systems](https://proceedings.neurips.cc/paper/2015/file/86df7dcfd896fcaf2674f757a2463eba-Paper.pdf).
+There is a long history of projects that end up in a disaster and that could be
+most likely saved by a proper data validation process.
+
+One of the common problems is a data distribution shift. In short, the data the
+model uses has significantly changed over time. This results in poor performance
+of an ML model. Therefore, detecting a data distribution shift is essential as
+soon as possible. There are several types of data distribution shifts. We will
+discuss two: a) covariate shift which we’ll cover later and b) concept drift,
+which you can see in the figure below. The figure shows the average number of
+rented bikes during non-working hours for the last two weekends of April and the
+first weekend in May (blue line) vs. the second weekend in May (orange line). We
+can see a significant change of pattern in the afternoon hours.
+
+![Example of concept drift](/uploads/images/2022-07-05/example_of_concept_drift.png '=800')
```

Comment:
`'=800' - @Meleagos what is the image initial size? this option should be used only to downsize. So, if you want to have 800px image, original one should have 1600px to look good on retina (4K) displays.